### PR TITLE
Auto tagger

### PR DESCRIPTION
**Pull request by cpefatimaabdillahi**
Hi,
- Please add ninjs_formatter_2 to vocabularies in superdesk-core/apps/prepopulate/data_init/vocabularies.json
Look for NINJSv3
`
{
        "_id": "subscriber_types",
        "display_name": "Subscriber Types",
        "type": "unmanageable",
        "selection_type": "do not show",
        "items": [
            {"is_active": true, "name": "All", "qcode": "all",
                "formats": [
                    {"name": "NINJS", "qcode": "ninjs"},
                    {"name": "NINJSv2", "qcode": "ninjs2"},
                    {"name": "NINJSv3", "qcode": "ninjs3"},
                    {"name": "NINJS FTP", "qcode": "ftp ninjs"},
                    {"name": "NewsML G2", "qcode": "newsmlg2"},
                    {"name": "Email", "qcode": "email"},
                    {"name": "Newsroom NINJS", "qcode": "newsroom ninjs"},
                    {"name": "Adobe InDesign (IDML)", "qcode": "idml"},
                    {"name": "iMatrics", "qcode": "imatrics"}
                ]},
            {"is_active": true, "name": "Digital/Internet", "qcode": "digital",
                "formats": [
                    {"name": "NINJS", "qcode": "ninjs"},
                    {"name": "NINJSv2", "qcode": "ninjs2"},
                    {"name": "NINJSv3", "qcode": "ninjs3"},
                    {"name": "NINJS FTP", "qcode": "ftp ninjs"},
                    {"name": "NewsML G2", "qcode": "newsmlg2"},
                    {"name": "Email", "qcode": "email"},
                    {"name": "Newsroom NINJS", "qcode": "newsroom ninjs"},
                    {"name": "Adobe InDesign (IDML)", "qcode": "idml"},
                    {"name": "iMatrics", "qcode": "imatrics"}
                ]},
            {"is_active": true, "name": "Wire/Paper", "qcode": "wire",
                "formats": [
                    {"name": "NITF", "qcode": "nitf"},
                    {"name": "Email", "qcode": "email"}
                ]}
        ],
        "init_version": 3
    }`

 
- Added jimi_2 file please use this version
- In cp_ninjs_formatter, it should use ninjs_formatter_2 (added it to the same directory) in the class not the current version from superdesk-core as we made changes to the main formatter

### CODE REVIEW

File: server/cp/output/formatter/jimi.py [hunk 1]
```diff
@@ -116,8 +116,8 @@ class JimiFormatter(Formatter):
 
     ENCODING = "utf-8"
 
-    type = "jimi"
-    name = "Jimi XML"
+    type = "jimi_2"
+    name = "Jimi XML 2"
```
Comment (petrjasek) [inline server/cp/output/formatter/jimi.py]: seems like you have `jimi_2` defined 2x now

File: server/cp/ai/__init__.py
Comment (petrjasek) [inline server/cp/ai/__init__.py:1]: why is this file needed in the repo?

File: server/cp/output/formatter/ninjs_formatter_2.py
Comment (petrjasek) [inline server/cp/output/formatter/ninjs_formatter_2.py:1]: would be good to only override the methods you need to change to avoid duplication

File: server/cp/output/formatter/ninjs_formatter_2.py [hunk 2]
```diff
@@ -0,0 +1,692 @@
...
+    * added ``evolvedfrom`` field
+    * added ``source`` field
+
+    Associations dictionary may contain entire items like
+    in `ninjs example <http://dev.iptc.org/ninjs-Examples-3>`_ or just the item ``guid``
+    and ``type``. In the latest case the items are sent separately before the package item.
+    """
+
+    type = "ninjs"
+    name = "NINJS"
```
Comment (petrjasek) [inline server/cp/output/formatter/ninjs_formatter_2.py]: would be good to have a custom name/type for this unless you want to replace existing formatter in current setup

File: server/cp/ai/__init__.py
Comment (cpefatimaabdillahi) [reply server/cp/ai/__init__.py:1]: This file is needed to handle the different read/write functions of the Semaphore.py file

File: server/cp/output/formatter/jimi.py (hunk 1 above)
Comment (cpefatimaabdillahi) [reply server/cp/output/formatter/jimi.py]: resolved in new update

File: server/cp/output/formatter/ninjs_formatter_2.py (hunk 2 above)
Comment (cpefatimaabdillahi) [reply server/cp/output/formatter/ninjs_formatter_2.py]: resolved with latest update

**Comment by cpefatimaabdillahi** [comment]
Hello @petrjasek , please see 6 new commits from Jan 30 and Jan 31. 

We have added the whole ninjs_formatter_2 and if you could help us override the methods it would be great.

Note For Semaphore.py:
Semaphore.py has a new variable called index_file_path  which refers to the path of a JSON file (Index.json) generated  when we create an Item in Metadata Management in SuperDesk. The file should be located where Semaphore is running i.e in /superdesk/server or /superdesk/server/data.

Please determine the path where this metadata file will be and set it in the environment variable index_file_path in the env file. I will share both files with Gideon. Thanks.

**Comment by petrjasek** [comment]
hi @cpefatimaabdillahi what's the structure of that file? would it make sense to store it as a vocabulary in db?

**Comment by petrjasek** [comment]
and which of all those PRs have latest code @cpefatimaabdillahi ? we should probably close the other 2

**Comment by cpefatimaabdillahi** [comment]
@petrjasek 189 and 190 can be closed.

**Comment by cpefatimaabdillahi** [comment]
> hi @cpefatimaabdillahi what's the structure of that file? would it make sense to store it as a vocabulary in db?

Hi @petrjasek,

Your suggestion makes sense. The JSON file we're discussing is nearly identical to the current vocabularies database, with the addition of semaphore codes necessary for mapping to IPTC codes. Would it be possible for you to adjust the script to read from the vocabularies database instead of a local file, which we've been using for testing? If so, could you please provide an estimate of the hours you anticipate this modification would take?

Thank you!

**Comment by petrjasek** [comment]
hi, I think you can just add those files here to begin with @cpefatimaabdillahi , in the data folder, and I can try to make a vocabulary from it

**Comment by pauljkelly5** [comment]
FWIW, I uploaded the vocab file manually to UAT. It's an update to the Index CV

I added a new property ("semaphore_id") to the schema which is present in all the items. It functions as a mapping to another CV in the manner of "ap_subject"
